cassandra/cassandra-docker.sh

```diff
@@ -96,43 +87,121 @@ if [ -n "$HELP" ]; then
   echo "        the classname to use as the seed provider"
   echo "        default: org.apache.cassandra.locator.SimpleSeedProvider"
   echo
-  echo "  --internode_encryption=[all|none|dc|rack]"
-  echo "        what type of internode encryption should be used"
-  echo "        default: none"
-  echo
-  echo "  --enable_client_encryption=[true|false]"
-  echo "        if client encryption should be enabled"
-  echo "        default: false"
+  echo "  --keystore-dir"
+  echo "        the location for the keystore and trust store files"
+  echo "        default: CASSANDRA_HOME/conf"
   echo
-  echo "  --require_node_auth=[true|false]"
-  echo "        if certificate based authentication should be required between nodes"
-  echo "        default: false"
+  echo "  --keystore-file"
+  echo "        the path to the keystore file"
+  echo "        default: CASSANDRA_HOME/conf/.keystore"
   echo
-  echo "  --require_client_auth=[true|false]"
-  echo "        if certificate based authentication should be required for client"
-  echo "        default: false"
+  echo "  --keystore-password"
+  echo "        the keystore password"
+  echo "        default: a random string"
   echo
-  echo "  --keystore_file=KEYSTORE_FILE_LOCATION"
-  echo "        the path to where the keystore is located"
+  echo "  --truststore-file"
+  echo "        the path to the trust store file"
+  echo "        default: CASSANDRA_HOME/conf/.truststore"
   echo
-  echo "  --keystore_password=KEYSTORE_PASSWORD"
-  echo "        the password to use for the keystore"
+  echo "  --truststore-password"
+  echo "        the truststore password"
+  echo "        default: a random string"
   echo
-  echo "  --keystore_password_file=KEYSTORE_PASSWORD"
-  echo "        a file containing only the keystore password"
+  echo "  --service-cert"
+  echo "        the cert for this service"
+  echo "        default: the tls.crt generated by OpenShift and mounted as secret"
   echo
-  echo "  --truststore_file=TRUSTSTORE_FILE_LOCATION"
-  echo "        the path to where the truststore is located"
+  echo "  --service-cert-key"
+  echo "        the cert key for this service"
+  echo "        default: the tls.key generated by OpenShift and mounted as secret"
   echo
-  echo "  --truststore_password=TRUSTSTORE_PASSWORD"
-  echo "        the password to use for the truststore"
-  echo
-  echo "  --truststore_password_file=TRUSTSTORE_PASSWORD"
-  echo "        a file containing only the truststore password"
+  echo "  --service-ca"
+  echo "        the CA file to include into the trust store"
+  echo "        default: the same as the service-cert"
+
   echo
   exit 0
 fi
 
+CASSANDRA_HOME=${CASSANDRA_HOME:-"/opt/apache-cassandra"}
```

Comment:
To understand these (and related) changes, it's good to know the following:

- The keystore should have only the keys needed by the server. For Cassandra, this means having the public/secret parts for the cert it's going to use in "server" mode, to talk to clients (proper clients or other nodes).
- Each trust level should use a different trust store. With this in mind, the trust store to be used with inter node communication should be different than the one used for client communication. 
- The inter-node trust store contains only the `/secrets/tls.crt` certificate, meaning that it accepts only nodes from the same service on the cluster.
- The client trust store contains the Service CA, which means that any certificates signed by this CA are accepted. In effect, this means that we'd be able to trust the Hawkular cert, if it would be possible to use that cert as a client cert
- As the certs generated by OpenShift can't be used for client auth, we set Cassandra's `client_auth` to false. This can be turned to true if the cert is changed to be allowed to be used as client cert.

Summary: 
- JGroups communication is encrypted
- Cassandra inter-node communication is encrypted
- Hawkular <-> Cassandra is encrypted
- There's no client auth happening (this might be a problem)